OpenKeychain/src/main/java/org/sufficientlysecure/keychain/pgp/CanonicalizedSecretKey.java

```diff
@@ -253,17 +247,33 @@ public PGPSignatureGenerator getCertSignatureGenerator(Map<ByteBuffer, byte[]> s
         }
     }
 
-    public PGPAuthenticationSignatureGenerator getAuthenticationSignatureGenerator(int hashAlgorithm,
-                                                                                   Map<ByteBuffer, byte[]> signedHashes)
+
+    private PGPContentSignerBuilder getAuthenticationContentSignerBuilder(int hashAlgorithm, Map<ByteBuffer,
+            byte[]> signedHashes) {
+        if (getAlgorithm() == PublicKeyAlgorithmTags.EDDSA) {
```

Comment:
This should probably also test for PRIVATE_KEY_STATE_DIVERT_TO_CARD in case of a key on a token, but I don't know how the Gnuk (or any other EdDSA-capable token) handle input data